# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1235540349 18000
#      Wed Feb 25 00:39:09 2009 -0500
# Node ID f71b749be1c1fc7efe49baf92772478abb268802
# Parent  31f86487724662b579c219abe9d4383a6e90ac32
input.cc (get_debug_input): don't pass arbitrary input to message as a format string

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,15 @@
+2009-02-25  John W. Eaton  <jwe@octave.org>
+
+	* input.cc (get_debug_input): Don't pass arbitrary input to
+	message as a format string.
+
+	* help.cc (__list_functions__): Simplify
+
 2009-02-24  John W. Eaton  <jwe@octave.org>
 
 	* help.cc, help.h (Vdoc_cache_file): New global variable.
 	(Fdoc_cache_file): New function.
 	* octave.cc (DOC_CACHE_FILE_OPTION): New option tag.
 	(long_opts): Include doc-cache-file in the list.
 	(octave_main): Handle DOC_CACHE_FILE_OPTION.
 
diff --git a/src/input.cc b/src/input.cc
--- a/src/input.cc
+++ b/src/input.cc
@@ -640,17 +640,17 @@ get_debug_input (const std::string& prom
 		buf << "\n" << curr_debug_line << ": " << line_buf;
 	    }
 	}
     }
 
   std::string msg = buf.str ();
 
   if (! msg.empty ())
-    message (Vgud_mode ? 0 : "keyboard", msg.c_str ());
+    message (Vgud_mode ? 0 : "keyboard", "%s", msg.c_str ());
 
   unwind_protect::begin_frame ("get_debug_input");
 
   unwind_protect_str (VPS1);
   VPS1 = prompt;
 
   while (Vdebugging)
     {
